lib/filter-stream.js

```diff
@@ -24,6 +24,7 @@ function FilterStream( blockMap, options ) {
   /** @type {Object} options */
   this.options = options || {}
 
+  this.options.readableObjectMode = true
```

Comment:
Bit of a minor nit, but I wonder if you should explicitly create a separate `transformOptions` object to pass into `stream.Transform.call` (and selectively set specific properties), rather than blindly passing in the `options` object passed into this function? (the latter might have side-effects, if somebody adds extra properties to the `options` object that you weren't expecting?)